Common/Global/GlobalVanillaItem.cs

```diff
@@ -0,0 +1,45 @@
+using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.ID;
+using Terraria.ModLoader;
+using System;
+using System.Collections.Generic;
+using TheBereftSouls.Common.Utility;
+
+namespace TheBereftSouls.Common.Global
+{
+	public class GlobalVanillaItem : GlobalItem
+	{
+		//Untested - JIT used to prevent crashes if Calamity isn't enabled.
+		
+
+		// Changes following Vanilla Weapons To Rogue
+		private static List<short> VanillaWeaponsToRogue = new List<short>
+		{
+        		ItemID.BoneJavelin,
+        		ItemID.ThrowingKnife,
+        		ItemID.Shuriken,
+	        	ItemID.BoneDagger,
+		};
+
+		[JITWhenModsEnabled("CalamityMod")]
+		public override void SetDefaults(Item item)
+		{
+			// Checks to see if Calamity are installed before making changes.
+			if (!ModLoader.TryGetMod("CalamityMod", out Mod CalamityMod))
+			{
+				return;
+			}
+			// Uses list to change all listed weapons to Rogue
+			foreach (int vanillaWeaponId in VanillaWeaponsToRogue)
+			{
+				if (item.type == vanillaWeaponId)
+				{
+					item.DamageType = GetModClass.rogueDamageClass;
+					base.SetDefaults(item);
```

Comment:
Gets called 4 times when it's in the foreach loop.

Reply:
> Gets called 4 times when it's in the foreach loop.

Well, my bad, not 4 times because of the if condition but writing it outside the for loop is still easier to understand.